### PR TITLE
Add the frecency-update ping

### PR DESCRIPTION
**Pull request by florian**
This pull request adds a new custom ping `frecency-update` for the [federated learning project](https://bugzilla.mozilla.org/show_bug.cgi?id=1462102).
The ping contains weight updates that are used to improve the global model we want to train.
There's also some other information added to evaluate the quality of the current model.

I built the project and made sure that all tests still pass.

### CODE REVIEW

**Comment by florian** [comment]
@acmiyaguchi Can you take a look at this?

File: schemas/telemetry/frecency-update/frecency-update.1.schema.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "properties": {
+    "client_id": {
```
Comment (acmiyaguchi) [inline schemas/telemetry/frecency-update/frecency-update.1.schema.json]: This is sent as `clientId` generally. You should use the template value `@TELEMETRY_CLIENTID_1_JSON@`, which includes the name and pattern.

File: schemas/telemetry/frecency-update/frecency-update.1.schema.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$",
+      "type": "string"
+    },
+    "payload": {
+      "properties": {
+        "loss": {
+          "type": "number"
+        },
+        "model_version": {
+          "type": "number"
```
Comment (acmiyaguchi) [inline schemas/telemetry/frecency-update/frecency-update.1.schema.json]: Out of curiosity, how often does the model version change and what is its purpose?

File: schemas/telemetry/frecency-update/frecency-update.1.schema.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        "update": {
+          "items": {
+            "type": "number"
+          },
+          "maxItems": 18,
+          "minItems": 18,
+          "type": "array"
+        }
+      },
+      "required": [
```
Comment (acmiyaguchi) [inline schemas/telemetry/frecency-update/frecency-update.1.schema.json]: This list should be alphabetized.

File: schemas/telemetry/frecency-update/frecency-update.1.schema.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    },
+    "type": {
+      "enum": [
+        "frecency-update"
+      ]
+    }
+  },
+  "required": [
+    "type",
+    "client_id",
```
Comment (acmiyaguchi) [inline schemas/telemetry/frecency-update/frecency-update.1.schema.json]: re: `clientId`

File: validation/telemetry/frecency-update.1.sample.pass.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{
+  "client_id": "e00d072c-db5c-42be-a159-a2e95466a7da",
```
Comment (acmiyaguchi) [inline validation/telemetry/frecency-update.1.sample.pass.json]: re: the browser reports `clientId`

File: schemas/telemetry/frecency-update/frecency-update.1.schema.json (hunk 3 above)
Comment (florian) [reply schemas/telemetry/frecency-update/frecency-update.1.schema.json]: Done.

File: schemas/telemetry/frecency-update/frecency-update.1.schema.json (hunk 2 above)
Comment (florian) [reply schemas/telemetry/frecency-update/frecency-update.1.schema.json]: The version tells the server what iteration the client based the update on. Every 30 minutes a new iteration is started. To apply the model updates, we need to know if the client computed them using the latest model version.

**Pull request merged**